sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/accessibility/AriaRoleCheck.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarSource HTML analyzer :: Sonar Plugin
+ * Copyright (c) 2010-2024 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.accessibility;
+
+import static org.sonar.plugins.html.api.HtmlConstants.ABSTRACT_ROLES;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.api.accessibility.AriaRole;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S6821")
+public class AriaRoleCheck extends AbstractPageCheck {
+    @Override
+    public void startElement(TagNode element) {
+        var role = element.getAttribute("role");
+        if (role == null) {
+            return;
+        }
+        var values = role.split(" ");
```

Comment:
can the role attribute have multiple values? if so, we should revisit implementation of other rules

Reply:
@ericmorand-sonarsource thoughts?

Reply:
seems that way - https://www.w3.org/TR/2014/REC-wai-aria-20140320/host_languages#host_general_role

```The attribute [value](https://www.w3.org/TR/2014/REC-wai-aria-20140320/terms#def_value) MUST allow a token list as the value;```

Reply:
Yes, this is confirmed by the specification of the attribute itself: https://www.w3.org/TR/role-attribute/#s_role_module_attributes

> The role attribute takes as its value one or more whitespace separated TERMorCURIEorAbsIRIs, which is defined in [[RDFA-CORE](https://www.w3.org/TR/role-attribute/#bib-RDFA-CORE)]